### ISSUE TITLE
int32 negative value turn into highest value int32 = 4294967295

### ISSUE THREAD

**Issue by u007** [grpc/grpc-node#1469]
### Problem description
int32 negative value turn into highest value int32 = 4294967295

### Reproduction steps
create a dart grpc server , and send a field with type int32 with value -1 or -16

### Environment
 - OS name, version and architecture: Ubuntu 18.04 
 - Node version v12.9.1
 - Node installation method nvm
 - If applicable, compiler version [e.g. clang 3.8.0-2ubuntu4]
 - Package name and version grpc 1.24.2

### Additional context
my dart running on same ubuntu 
Dart VM version: 2.8.2 (stable)
compiled via dart2native on Ubuntu 20.04 LTS
Package: GRPC 2.0.3

**Comment by murgatroid99** [comment]
Are you using the Dart gRPC library? If so, I think your issue belongs in https://github.com/grpc/grpc-dart.

**Comment by u007** [comment]
yes, but when i use the dart library to retrieve from dart server, it works properly.
but when on nodejs, the int32 behave that way. double works properly thou...
do i have to file this bug in dart grpc?

**Comment by murgatroid99** [comment]
I misunderstood your initial issue. It looks like you're talking about a Protobuf problem. What protobuf library are you using with your Node.js clients?

**Comment by u007** [comment]
google-protobuf: "3.12.2"

**Comment by murgatroid99** [comment]
I am guessing that that means that you generated the code using the `grpc-tools` package. Can you try updating to the latest version of that library and re-generating the code. There was a relatively recent release with a newer version of the code generator.

If that doesn't help, `google-protobuf` and `protoc` are maintained in this repository: https://github.com/protocolbuffers/protobuf. That is probably the best place to file an issue about this. When you do I suggest sharing the contents of the `.proto` files, or at least the message type(s) that you are having problems with, so that they can see the problem more specifically.

**Comment by u007** [comment]
iam using grpc-tools 1.9.0, ok i will try to raise ticket there
